executor/runtime/docker/docker.go

```diff
@@ -1588,7 +1588,7 @@ func (r *DockerRuntime) createAllExtraContainers(ctx context.Context, pod *v1.Po
 	for idx := range r.c.ExtraUserContainers() {
 		c := r.c.ExtraUserContainers()[idx]
 		group.Go(func(ctx context.Context) error {
-			cid, err := r.createExtraContainerInDocker(ctx, c.V1Container, mainContainerID, mainContainerRoot)
+			cid, err := r.createExtraContainerInDocker(ctx, c.V1Container, mainContainerID, mainContainerRoot, *pod)
```

Comment:
Why not just pass by pointer here?

Reply:
I've fixed that in all these related function so it is always the pointer now.
I'm a tad paranoid about mutating the pod when I don't mean to.